### PR TITLE
`d\managed_disk`: Add support for `encryption_settings`

### PR DESCRIPTION
**Pull request by myc2h6o**
`enabled` is to be deprecated per https://github.com/hashicorp/terraform-provider-azurerm/issues/14574#issuecomment-992690956 so not added in the doc

### CODE REVIEW

File: internal/services/compute/managed_disk_data_source.go [hunk 1]
```diff
@@ -55,6 +55,54 @@ func dataSourceManagedDisk() *pluginsdk.Resource {
 				Computed: true,
 			},
 
+			"encryption_settings": {
```
Comment (katbyte) [inline internal/services/compute/managed_disk_data_source.go:58]: settings is redundant in property names
```suggestion
			"encryption": {
```

Comment (myc2h6o) [reply internal/services/compute/managed_disk_data_source.go:58]: in snapshot and managed disk resource it is already named `encryption_settings`. Maybe let's keep it consistent with them?
https://github.com/hashicorp/terraform-provider-azurerm/blob/2443a8844acc475bff373d12ba4858b06561f126/internal/services/compute/managed_disk_resource.go#L187
https://github.com/hashicorp/terraform-provider-azurerm/blob/2443a8844acc475bff373d12ba4858b06561f126/internal/services/compute/snapshot_resource.go#L88

**Review by stephybun — CHANGES_REQUESTED** [review]
We also have a test failure
```
------- Stdout: -------
=== RUN   TestAccDataSourceManagedDisk_encryptionSettings
=== PAUSE TestAccDataSourceManagedDisk_encryptionSettings
=== CONT  TestAccDataSourceManagedDisk_encryptionSettings
testing_new.go:70: Error running post-test destroy, there may be dangling resources: exit status 1
Error: purging of Secret "secret-bf1zk" (Key Vault "https://acctestkv-bf1zk.vault.azure.net/") : keyvault.BaseClient#PurgeDeletedSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=*******;oid=3aa04c8c-5a75-4e5e-9117-1b7cf6f33e21;numgroups=9;iss=https://sts.windows.net/*******/' does not have secrets purge permission on key vault 'acctestkv-bf1zk;location=westeurope'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"ForbiddenByPolicy"}
--- FAIL: TestAccDataSourceManagedDisk_encryptionSettings (289.24s)
FAIL
```

**Comment by myc2h6o** [comment]
Hi @stephybun thanks for pointing out the issue. I've fixed the test case and rerun the test cases, please take a look, thanks!
image

**Review by stephybun — COMMENTED** [review]
Thanks @myc2h6o, the name should still be shortened to `encryption` since `_settings` is implied for the block. This block should be renamed in the other resources as well.

**Comment by myc2h6o** [comment]
Got it @stephybun I've renamed it to `encryption` and later I'll update the name in existing resources in a separate pr. The vendor dependency seems breaking, I'll merge main after it's fixed

**Comment by myc2h6o** [comment]
Updated test result
image

File: internal/services/compute/managed_disk_data_source.go (hunk 1 above)
Comment (tombuildsstuff) [reply internal/services/compute/managed_disk_data_source.go]: 👍 works for me

**Comment by myc2h6o** [comment]
Hi @tombuildsstuff thanks for reviewing the pr! I've reverted the name back to `encryption_settings`, could you take another look?
Updated test result:
image

**Review by katbyte — APPROVED** [review]
LGTM 🎩

**Pull request merged**